### PR TITLE
[W4.4b] [T11-A1] Sarah Goh Shi Ning

### PR DESCRIPTION
**Pull request by sarahgoh97**
Added Block, Street, Unit and PostalCode classes

### CODE REVIEW

**Comment by sarahgoh97** [comment]
Updated test cases

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -57,3 +80,60 @@ public boolean isPrivate() {
         return isPrivate;
     }
 }
+/**
+ * Represents a Person's address's block number in the address book.
+ */
+class Block {
+    private int blockNumber;
```
Comment (ashrafabdul) [inline src/seedu/addressbook/data/person/Address.java:67]: You should use getters and setters to for storing and retrieving values from class members. The constructor is ideally for initialization operations only.

File: src/seedu/addressbook/Main.java [hunk 2]
```diff
@@ -105,12 +105,16 @@ private void recordResult(CommandResult result) {
...
-        try {
-            command.setData(addressBook, lastShownList);
-            CommandResult result = command.execute();
+        command.setData(addressBook, lastShownList);
+        CommandResult result = command.execute();
+        try{
             storage.save(addressBook);
             return result;
+        } catch(StorageOperationException e){
+            ui.showToUser(e.getMessage());
```
Comment (ashrafabdul) [inline src/seedu/addressbook/Main.java:116]: As discussed in class, the control goes to showToUser method that prints a friendly error message. Is there something else we can do by handling exceptions more gracefully? Maybe save the program state to give the user a chance to change the file permission?

File: src/seedu/addressbook/Main.java [hunk 3]
```diff
@@ -105,12 +105,16 @@ private void recordResult(CommandResult result) {
      * @param command user command
      * @return result of the command
      */
+    //credit to #111, Cho Chih Tun for helping me and explaining
     private CommandResult executeCommand(Command command)  {
-        try {
-            command.setData(addressBook, lastShownList);
-            CommandResult result = command.execute();
```
Comment (ashrafabdul) [inline src/seedu/addressbook/Main.java:111]: Is there a reason to not include these in the try block? What exceptions can they generate and how will they be handled?

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -25,8 +32,24 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
...
+        this.value = block.getBlock() + COMMA + street.getStreet() + COMMA + unit.getUnit() + COMMA + postalCode.getPostalCode();
+    }
+
+    /**
+     * Breaking up the address into its attributes: Block, Street, Unit and PostalCode
+     * @param address
+     */
+    private void breakUpAddress(String address) {
+        int firstCommaIndex = address.indexOf(COMMA);
+        block = new Block(address.substring(0,firstCommaIndex));
```
Comment (ashrafabdul) [inline src/seedu/addressbook/data/person/Address.java]: In this case you should be using getters and setters as discussed in class.

**Review by ashrafabdul — COMMENTED** [review]
Good job on following up from class discussion by setting appropriate access modifier for your class members. however you should be updating them through getter and setters. Follow the coding standard at https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html

Also, these should have been two different PRs, one for the exception  handling and the other for updating the Address class. accepting this for now but please ensure clean PRs going forward.

**Pull request closed without merging**